csp_billing_adapter_local/plugin.py

```diff
@@ -33,6 +33,27 @@
 ADAPTER_DATA_DIR = '/var/lib/csp-billing-adapter'
 CACHE_FILE = 'cache.json'
 CSP_CONFIG_FILE = 'csp-config.json'
+CSP_LOG_FILEPATH = '/var/log/csp_billing_adapter.log'
+
+
+def _set_logger_file():
+    """Set logger file with level and format."""
+    try:
+        local_config_path = Path(ADAPTER_DATA_DIR).joinpath(CSP_CONFIG_FILE)
```

Comment:
if we wanted to reuse code (I'm in favour),
the log setting should be at the bottom of the file, which is not where we usually set the log

```python
def _set_logger_file():
    """Set logger file with level and format."""
    csp_config = get_csp_config(Config({}))
    csp_config_logging = csp_config.get('logging', {})
    logging.basicConfig(
        filename=CSP_LOG_FILEPATH,
        level=csp_config_logging.get('level', 'INFO'),
        format="%(asctime)s %(levelname)s: %(message)s"
    )
    
    _set_logger_file()
```

Thoughts ?

Reply:
The above snippet results in a recursive call of the function to itself, infinitely... Maybe you meant to call to _set_logger_file() to be outdented one level?

Reply:
Fixed the indentation